azkaban-common/src/main/java/azkaban/utils/Emailer.java

```diff
@@ -46,8 +48,9 @@
   private final String tls;
   private boolean testMode = false;
 
-  public Emailer(final Props props) {
+  public Emailer(final Props props, final CommonMetrics commonMetrics) {
     super(props);
+    this.commonMetrics = commonMetrics;
```

Comment:
I'm revisiting this design. I was thinking if we should have requireNonNull to wrap commonMetrics. However, if we do that way, other program (external MP) can not use Emailer easily. I'm thinking if we could let contructor doesn't include the `commonMetrics`? Does it make sense?